src/interface/mesh_data.hpp

```diff
@@ -289,50 +289,211 @@ class MeshData {
+                                     PackIndexMap &map) {
+    return PackVariablesAndFluxesImpl(&map, nullptr, names_or_flags, sparse_ids);
+  }
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
+                                     const std::vector<int> &sparse_ids) {
+    return PackVariablesAndFluxesImpl(nullptr, nullptr, names_or_flags, sparse_ids);
+  }
+  // no sparse ids
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
+                                     PackIndexMap &map, vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(&map, &key, names_or_flags);
+  }
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
+                                     vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(nullptr, &key, names_or_flags);
   }
-
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args, PackIndexMap &map,
-                            std::vector<std::string> &key) {
-    return PackVariablesImpl(&map, &key, std::forward<Args>(args)...);
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
+                                     PackIndexMap &map) {
+    return PackVariablesAndFluxesImpl(&map, nullptr, names_or_flags);
   }
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args, PackIndexMap &map) {
-    return PackVariablesImpl(&map, nullptr, std::forward<Args>(args)...);
+  template <typename Elem>
+  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags) {
+    return PackVariablesAndFluxesImpl(nullptr, nullptr, names_or_flags);
   }
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args, std::vector<std::string> &key) {
-    return PackVariablesImpl(nullptr, &key, std::forward<Args>(args)...);
+  // only sparse ids
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids,
+                                     PackIndexMap &map, vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(&map, &key, sparse_ids);
   }
-  template <typename... Args>
-  const auto &PackVariables(Args &&... args) {
-    return PackVariablesImpl(nullptr, nullptr, std::forward<Args>(args)...);
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids,
+                                     vpack_types::StringPair &key) {
+    return PackVariablesAndFluxesImpl(nullptr, &key, sparse_ids);
+  }
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids,
+                                     PackIndexMap &map) {
+    return PackVariablesAndFluxesImpl(&map, nullptr, sparse_ids);
+  }
+  const auto &PackVariablesAndFluxes(const std::vector<int> &sparse_ids) {
+    return PackVariablesAndFluxesImpl(nullptr, nullptr, sparse_ids);
+  }
```

Comment:
Are these actually required?
If I'm not missing something they're matching the
```c++
  const auto &PackVariablesAndFluxes(const std::vector<Elem> &names_or_flags,
                                     PackIndexMap &map, vpack_types::StringPair &key)
```
above, aren't they?

Reply:
These exclude either the `PackIndexMap` or the key, or both. Whereas the one you quoted has both of them. They're just syntactic sugar overloads.